mmv1/products/firebasedatabase/api.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: FirebaseDatabase
+display_name: Firebase Database
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://firebasedatabase.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/firebase
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Instance'
+    min_version: beta
+    base_url:  projects/{{project_id}}/locations/{{location_id}}/instances
+    self_link: projects/{{project_id}}/locations/{{location_id}}/instances/{{database_id}}
+    create_url:  projects/{{project_id}}/locations/{{location_id}}/instances?databaseId={{database_id}}
+    input: true
+    description: |
+      A Firebase Realtime Database Instance
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://firebase.google.com/'
+      api: 'https://firebase.google.com/docs/reference/rest/database/database-management/rest'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource name of the Database Instance.
+        output: true
+        pattern: projects/{{project}}/locations/{{location_id}}/instances/{{database_id}}
+      - !ruby/object:Api::Type::String
+        name: 'projectId'
```

Comment:
It looks like the URL uses the project number rather than the project id.

Reply:
It looks like there is also a separate `project` property, which is not a url param. https://firebase.google.com/docs/reference/rest/database/database-management/rest/v1beta/projects.locations.instances#DatabaseInstance

Reply:
The URL actually accepts both the project Id or the Project Number in the same place, and as you noticed there's a separate project property that is taking what seems to be the most canonical 'project' url param name.

I was thinking that 'projectId' was a better name to represent that it could accept both. Looking at https://google.aip.dev/cloud/2510 it seems to prefer project number, but that seems like it would be inconsistent with what I'm able to find other providers doing.

WDYT?

Reply:
sorry for the delay - this slipped through the cracks. Looking at this again - I think it might make sense to pattern this off of Memcache Instance: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/products/memcache/api.yaml#L43

i.e. rather than having a projectId, can you put `{{project}}` in the url and let magic modules automatically generate the project field for you?

Reply:
I don't see any indication in the [API docs](https://firebase.google.com/docs/reference/rest/database/database-management/rest/v1beta/projects.locations.instances/create) that this accepts project id as well as project number, so I don't think we need to support that use case.